src/main/scala/com/microsoft/hyperspace/index/rankers/FilterIndexRanker.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (2020) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.index.rankers
+
+import com.microsoft.hyperspace.index.IndexLogEntry
+
+/**
+ * Ranker class for Filter rule indexes
+ */
+object FilterIndexRanker {
+
+  /**
+   * Pick the best index among all available index options according to their cost.
+   *
+   * If hybridScanEnabled is true, pick the index which covers the most source data
+   * so that we could minimize the amount of data for on-the-fly shuffle or merge.
+   * Otherwise, return the head of the given index list.
```

Comment:
We are not really doing shuffle for filter index, right? You may need to document a different reasoning for getting the max number of source files.

Reply:
Yes, for filter we don't shuffle the data (though the current version do shuffle in case non-parquet format). I'll try to rewrite the comment.